diff --git a/home/adam/vcs/git/t/t1517-outside-repo.sh b/t/t1517-outside-repo.sh
index c824c1a..853ab10 100755
--- a/home/adam/vcs/git/t/t1517-outside-repo.sh
+++ b/t/t1517-outside-repo.sh
@@ -115,7 +115,7 @@ do
 	credential-osxkeychain | cvsexportcommit | cvsimport | cvsserver | \
 	daemon | \
 	difftool--helper | filter-branch | fsck-objects | get-tar-commit-id | \
-	gui | gui--askpass | \
+	gui | gui--askpass | gui--askyesno | \
 	http-backend | http-fetch | http-push | init-db | \
 	merge-octopus | merge-one-file | merge-resolve | mergetool | \
 	mktag | p4 | p4.py | pickaxe | remote-ftp | remote-ftps | \
