diff --git a/t/t3900-i18n-commit.sh b/t/t3900-i18n-commit.sh
index f27d09cfd9..aefcbc5128 100755
--- a/t/t3900-i18n-commit.sh
+++ b/t/t3900-i18n-commit.sh
@@ -110,13 +110,20 @@ test_expect_success 'ISO8859-1 should be shown in UTF-8 now' '
 	compare_with ISO8859-1 "$TEST_DIRECTORY"/t3900/1-UTF-8.txt
 '
 
-for H in eucJP ISO-2022-JP
+for H in eucJP
 do
 	test_expect_success "$H should be shown in UTF-8 now" '
 		compare_with '$H' "$TEST_DIRECTORY"/t3900/2-UTF-8.txt
 	'
 done
 
+for H in ISO-2022-JP
+do
+	test_expect_failure "$H should be shown in UTF-8 now" '
+		compare_with '$H' "$TEST_DIRECTORY"/t3900/2-UTF-8.txt
+	'
+done
+
 test_expect_success 'config to add customization' '
 	git config --unset-all i18n.commitencoding &&
 	if Z=$(git config --get-all i18n.commitencoding)
@@ -144,13 +151,20 @@ test_expect_success 'ISO8859-1 should be shown in UTF-8 now' '
 	compare_with ISO8859-1 "$TEST_DIRECTORY"/t3900/1-UTF-8.txt
 '
 
-for H in eucJP ISO-2022-JP
+for H in eucJP
 do
 	test_expect_success "$H should be shown in UTF-8 now" '
 		compare_with '$H' "$TEST_DIRECTORY"/t3900/2-UTF-8.txt
 	'
 done
 
+for H in ISO-2022-JP
+do
+	test_expect_failure "$H should be shown in UTF-8 now" '
+		compare_with '$H' "$TEST_DIRECTORY"/t3900/2-UTF-8.txt
+	'
+done
+
 for J in eucJP ISO-2022-JP
 do
 	if test "$J" = ISO-2022-JP
@@ -209,7 +223,12 @@ test_commit_autosquash_multi_encoding () {
 	old=$2
 	new=$3
 	msg=$4
-	test_expect_success "commit --$flag into $old from $new" '
+	if [[ "$flag" = fixup && "$old" = ISO-2022-JP && "$new" = UTF-8 ]]; then
+		cmd=test_expect_failure
+	else
+		cmd=test_expect_success
+	fi
+	"$cmd" "commit --$flag into $old from $new" '
 		git checkout -b $flag-$old-$new C0 &&
 		git config i18n.commitencoding $old &&
 		echo $old >>F &&
